.github/workflows/pull-request.yml

```diff
@@ -27,81 +29,81 @@ env:
 
 jobs:
 
-  supported-reqs:
+  build-basic:
 
-    name: 'Supported scripts/requirements.txt'
+    name: "build with requirements.txt"
     runs-on: ubuntu-20.04
 
     steps:
       - uses: actions/checkout@v2
 
-      - name: apt-get install
-        run: sudo apt-get -y install $ubuntu_base_deps
-
-      - name: PATH += .local/bin
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
+      - name: setup env with requirements.txt
+        run: |
+          sudo apt-get -y install $ubuntu_base_deps
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          pip install -r scripts/requirements.txt
 
-      - name: 'pip install -r scripts/requirements.txt'
-        run: pip install -r scripts/requirements.txt
+      - name: build doc
+        run: ./.github/actions/build-sof-doc.sh
 
-      # No YAML anchors; this is copy/paste
-      - name: configure SOF doc
+      - name: prepare file for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
         run: |
-          git clone https://github.com/thesofproject/sof
-          cmake -GNinja -S sof/doc -B _build_doxy
+          mkdir _build/latest
+          mv -v _build/html/* _build/latest
+          mv _build/latest _build/html/
+          touch _build/html/.nojekyll
+          mv scripts/publish-README.md _build/html/README.md
+          mv scripts/publish-index.html _build/html/index.html
```

Comment:
Thanks for the `build-sof-doc.sh` script, reduces duplication and makes it possible to test locally. Please create a script for this section too so it can tested locally too.

Reply:
I am not sure if this is needed as we will only use these scripts once. The local action is replace for the missing YAML anchors support I think.

Reply:
Did you read my comment?

Reply:
maybe I misunderstand your here, so you want some script to run **locally**, but not a local action?

Reply:
but why we need a script to run locally?

Reply:
For the ability to test  as much as possible before git pushing of course. Just a suggestion.

Reply:
but these bash only did some file move for the deploy. To open the HTML, you only need to open the `index.html` in  `_build/html/`

Reply:
If someone changes this script, they cannot really test their change before pushing to github and they cannot look at the resulting directories either.

Reply:
Of course, this script can only be test in GitHub Action ENV. And that's why I played with it in my own forked repo first.

Reply:
Full testing requires Github Actions. Moving a few files and directories and looking at the result can and should be done locally first, it's much more efficient. Everything that can be done locally should be done locally.